acct-user/openntpd/openntpd-0.ebuild

```diff
@@ -0,0 +1,14 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit acct-user
+
+DESCRIPTION="user for openntpd daemon"
+ACCT_USER_ID=321
+ACCT_USER_GROUPS=( openntpd )
+ACCT_USER_HOME=/var/lib/openntpd/chroot
+ACCT_USER_HOME_OWNER=root:root
```

Comment:
These look a bit suspicious. To my understanding, HOME_OWNER is defined if there's some other package using that directory. Also, is `/chroot` necessary? Won't the defaults work?

Rest about this PR looks good. We just want to get acct ebuilds into good condition before pushing, because modifying them later is annoying.

Reply:
The default for a home directory is /dev/null, that won't work. openntpd requires a directory to chroot to:

https://github.com/openntpd-portable/openntpd-portable/blob/master/INSTALL
"This requires that a user, group and directory to be created for it.
The user should not be permitted to log in, and its home directory
should be owned by root and be mode 755."

It also needs a place to store the ntpd.drift file:

$ ls -l /var/lib/openntpd/
total 8
drwxr-xr-x 2 root root 4096 Sep 26 17:33 chroot
-rw-r--r-- 1 root root    6 Dec  2 09:39 ntpd.drift

So the /var/lib/openntpd directory is where the privileged piece stores stuff, and the chroot subdirectory is where the unprivileged piece chroots to.

By default it would write the drift file to /var/db, the ebuild moves it to /var/lib/openntpd. I guess if we left it in /var/db we could make the homedir just /var/lib/openntpd without the chroot, but then the drift file is mixed in with other stuff and we'd need the ebuild to move existing drift files and whack the chroot subdir on existing systems.

Overall I think this is the cleanest approach but I'm up for trying alternatives if need be.

Thanks...

Reply:
Mmhm, sounds good to me. 
@orlitzky any comments?

Reply:
I looked at this on bug 693050 and openntpd is indeed an exception. The daemon always chroots to the user's home directory, and that directory has to be owned by root (the daemon will refuse to start), so you need to set both of those variables in the acct-user ebuild.

So long as `/var/lib/openntpd` is also owned by root (and not openntpd), the use of  `/var/lib/openntpd/chroot` sounds good to me. But if `openntpd` owns the parent directory, having `/var/lib/openntpd/chroot` be owned by root doesn't add any security.

Reply:
On existing installs that is true:

# ls -ld /var/lib/openntpd
drwxr-xr-x 3 root root 4096 May 20  2019 /var/lib/openntpd
# ls -ld /var/lib/openntpd/chroot
drwxr-xr-x 2 root root 4096 Sep 26 17:33 /var/lib/openntpd/chroot

I removed the package completely including removing those directories, and a fresh install looks good too:

# ls -ld /var/lib/openntpd
drwxr-xr-x 3 root root 4096 Dec  3 15:15 /var/lib/openntpd
# ls -ld /var/lib/openntpd/chroot
drwxr-xr-x 2 root root 4096 Dec  3 15:15 /var/lib/openntpd/chroot